ui_testing/testcases/basic_tests/test04_testunits.py

```diff
@@ -1058,6 +1058,61 @@ def test030_allow_unit_field_to_be_displayed_in_case_of_mibi(self):
             self.base_selenium.LOGGER.info('search for value of the unit field: {}'.format(row_data['Unit']))
             self.assertIn(row_data['Unit'], fixed_sheet_row_data)
 
+    def test032_edit_category_edits_category_label_in_testplan_step_two(self):
+        """
+
+        New: Test unit: Category Approach: Any update in test unit category should reflect in the test plan ( step two ) in this test unit
+
+        LIMS-3687
+        :return:
+        """
+        new_random_name = self.generate_random_string()
+        new_random_method = self.generate_random_string()
+        new_random_category = self.generate_random_string()
+
+        self.base_selenium.LOGGER.info('Create new testunit with qualitative and random generated data')
+        self.test_unit_page.create_qualitative_testunit(name=new_random_name, method=new_random_method,
+                                                        material_type='All', unit='',
+                                                        category=new_random_category)
+
+        self.test_unit_page.sleep_tiny()
+        self.test_unit_page.save(save_btn='general:save_form', logger_msg='Save new testunit')
+
+        active_testplan_data = self.get_active_article_with_tst_plan()
+        self.test_plan.get_test_plans_page()
+        self.base_selenium.LOGGER.info('Create new testPlan to use the newly created testunit')
+        new_random_testplan_name = self.test_plan.create_new_test_plan(test_unit=new_random_name,
+                                                                       material_type=
+                                                                       active_testplan_data['Material Type'],
+                                                                       article=
+                                                                       active_testplan_data['Article Name'])
```

Comment:
If Mazan developed the API, this could be replaced

Reply:
@0xIslamTaha create APIs not developed yet, and actually today is my last full day with 1LIMS(not even sure if it will be the full day), so I think this is the better solution for now.

Reply:
done